clients/src/main/java/org/apache/kafka/clients/CommonClientConfigs.java

```diff
@@ -219,6 +219,19 @@ public class CommonClientConfigs {
     public static final String DEFAULT_API_TIMEOUT_MS_DOC = "Specifies the timeout (in milliseconds) for client APIs. " +
             "This configuration is used as the default timeout for all client operations that do not specify a <code>timeout</code> parameter.";
 
+    public static final String METADATA_RECOVERY_STRATEGY_CONFIG = "metadata.recovery.strategy";
+    public static final String METADATA_RECOVERY_STRATEGY_DOC = "Controls how the client recovers when none of the brokers known to it is available. " +
+            "If set to <code>none</code>, the client fails. If set to <code>rebootstrap</code>, " +
+            "the client repeats the bootstrap process using <code>bootstrap.servers</code>. " +
+            "Rebootstrapping is useful when a client communicates with brokers so infrequently " +
+            "that the set of brokers may change entirely before the client refreshes metadata. " +
+            "Opportunities to rebootstrapping depend on connection establishing and reconnect timeouts and the broker count. " +
+            "The timeouts may prevent identifying brokers as unavailable simultaneously, which is necessary to trigger rebootstrapping. " +
```

Comment:
This section is very confusing for me, and I think this should be reworded.

Maybe it can flow like this: "Metadata recovery is triggered when all last-known brokers appear unavailable simultaneously. Brokers appear unavailable when disconnected and no current retry attempt is in-progress."

Reply:
This makes sense. Thank you, updated the doc